### PR TITLE
using ipython to convert ipython magic issue: #26294

### PR DESCRIPTION
**Pull request by NikolaMandic**
hey will this work for issue #26294?

### CODE REVIEW

**Comment by NikolaMandic** [comment]
I've signed it

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @dynamicwebpaige: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/tools/compatibility/ast_edits.py [hunk 1]
```diff
@@ -544,8 +544,11 @@ def process_opened_file(self, in_filename, in_file, out_filename, out_file):
       A tuple representing number of files processed, log of actions, errors
     """
     lines = in_file.readlines()
+    import IPython.core.inputtransformer2 as ip
```
Comment (us) [inline tensorflow/tools/compatibility/ast_edits.py]: please import on top

**Review by us — CHANGES_REQUESTED** [review]

File: tensorflow/tools/compatibility/ast_edits.py (hunk 1 above)
Comment (NikolaMandic) [reply tensorflow/tools/compatibility/ast_edits.py]: did that now

File: tensorflow/tools/compatibility/ast_edits.py [hunk 2]
```diff
@@ -29,6 +29,8 @@
 import pasta
 import six
 
+import IPython.core.inputtransformer2 as ip
```
Comment (martinwicke) [inline tensorflow/tools/compatibility/ast_edits.py]: I'd rather avoid a hard dependency. Can you put this import into a try block and make the processing based on it conditional on it succeeding?

**Review by martinwicke — CHANGES_REQUESTED** [review]

File: tensorflow/tools/compatibility/ast_edits.py (hunk 2 above)
Comment (NikolaMandic) [reply tensorflow/tools/compatibility/ast_edits.py]: I just adde change to my master I guess it needs some time to update on pull request

**Comment by martinwicke** [comment]
Looks good to me. @lc0 is there enough test coverage to be comfortable after tests pass or do we need to test this manually?

**Review by martinwicke — DISMISSED** [review]

**Comment by lc0** [comment]
The issues is 3 month old. Does anybody has an example of test that fails currently? @NikolaMandic 

The simple magic should be working fine, that what was tested in previous PRs as well

cc @martinwicke

**Comment by lc0** [comment]
Also, unfortunately my changes I tested with a set of test notebooks, but these never made to the repo. Was also not completely sure where to store the fixtures.

If you point me out what is the preferred way to add such tests, I would add my old test cases.

**Comment by martinwicke** [comment]
I would simply add a testdata folder right there, containing your test notebooks (which are hopefully very small). And then add a test that runs the converter on them.

**Comment by NikolaMandic** [comment]
@lc0 I just tested it manually

**Comment by lc0** [comment]
> @lc0 I just tested it manually the original issue was to convert magics in notebooks so it did not work since there was complaint about it

@NikolaMandic do you have an example, when it fails with current nightly? 

@martinwicke I am going to add these test files over the weekend

**Review by Meathead98 — CHANGES_REQUESTED** [review]
Y

**Comment by NikolaMandic** [comment]
@lc0 hm I just saw test that failed lets see if it will pass after correction I pushed now

**Comment by lc0** [comment]
@NikolaMandic one more time, would you mind to provide an example where you magic does not work?

**Comment by lc0** [comment]
> I would simply add a testdata folder right there, containing your test notebooks (which are hopefully very small). And then add a test that runs the converter on them.

I am going to add my test decks over the weekend so we have a bit better setup.

With a proposed change itself, I am still not sure what case we are trying to solve. And currently it looks like we do not have dependency, so we check `globals` and at the end of the day do nothing, since it's not imported

**Comment by NikolaMandic** [comment]
@lc0 I understood it as someone in tensorflow team wrote an actual parser to convert python files and issue with it was that it could not handle magics so this converts magics to python code before feeding that parser. This is original issue as I saw it.

**Review by alextp — APPROVED** [review]

**Comment by lc0** [comment]
@alextp the issue with magic was handled way before this. So far I can not see an example to reproduce a bug.

By merging this I am not sure what we are fixing.

**Comment by lc0** [comment]
Additionally I do not see `Ipython` as a dependency, so it feels a bit of like a dead code, that never gets executed. Correct me if I am wrong cc @alextp

**Comment by alextp** [comment]
You're right, let's just close this.

**Comment by lc0** [comment]
Do we need to remove `ready to pull` tag, or would copybara pick it up since it has a tag even when the PR is closed? cc @alextp

**Pull request closed without merging**